shell-ui/src/auth/permissionUtils.js

```diff
@@ -0,0 +1,40 @@
+//@flow
+import type { Options, TranslationAndGroups } from '../index';
+
+export const isEntryAccessibleByTheUser = (
+  [path, translationAndGroup]: [string, TranslationAndGroups],
+  userGroups: string[],
+): boolean => {
+  return (
+    (translationAndGroup.groups &&
+      translationAndGroup.groups
+        .map((group) => userGroups.includes(group))
+        .reduce(
+          (aggregatedPredicates, current) => aggregatedPredicates && current,
+          true,
+        )) ||
+    !translationAndGroup.groups
+  );
+};
+
+export const getAccessiblePathsFromOptions = (
```

Comment:
Is this method ever used (outside of tests :sweat_smile:)?

Reply:
Ah, I was reading through the commits, and it's used only for pre-fetching all accessible paths.